tests/unit/lxd/test_launcher.py

```diff
@@ -41,22 +42,35 @@ def mock_lxc():
 
 
 @pytest.fixture
-def mock_lxd_instance():
-    with mock.patch(
+def fake_instance():
+    """Returns a fake LXD Instance"""
+    instance = MagicMock()
+    # the name has an invalid character to ensure the instance_name will be different
+    # so the two are not conflated in the unit tests
+    instance.name = "test-instance-$"
+    instance.instance_name = "test-instance-fa2d407652a1c51f6019"
```

Comment:
I'm not familiar with the internal structure of instances but `instance.name` and `instance.instance_name` sound a bit confusing. Just mentioning, I don't think there's much we can do about that here in tests.

Reply:
I agree with you, it's a confusing subtlety that has arisen from craft-provider's evolving without breaking changes.